waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -803,9 +809,10 @@
         ;; delay iff the log-bucket-url setting was given the scheduler config.
         log-bucket-sync-secs (if log-bucket-url (:log-bucket-sync-secs context) 0)
         total-sigkill-delay-secs (+ pod-sigkill-delay-secs log-bucket-sync-secs)
+        has-reverse-proxy? (get env "GRPC_TRANSCODER")
         ;; Make $PORT0 value pseudo-random to ensure clients can't hardcode it.
         ;; Helps maintain compatibility with Marathon, where port assignment is dynamic.
-        port0 (-> service-id hash (mod 100) (* 10) (+ pod-base-port))
+        port0 (generate-port-with-offset service-id pod-base-port (if has-reverse-proxy? 1 0))
```

Comment:
Please have the `1` available as a constant and make usable `generate-port-with-offset`  in this function and `pod->ServiceInstance`.